src/query.rs

```diff
@@ -195,6 +195,12 @@ impl Query {
         self.client.add_option(name, value);
         self
     }
+
+    pub fn with_param(self, name: &str, value: impl Serialize) -> Result<Self, String> {
```

Comment:
`Result<>` can be very inconvenient. It can be used in a concise way only inside `-> Result` functions (`q.with_param("a", 42)?.with_param("b", 43)?`), but sometimes you want explicitly handle errors with matching inside the function that doesn't return `Result`, so `?` (via the `Try` trait) cannot be used.

So, I suggest delayed errors in the same way as in the `bind()` method.

Reply:
seems like it'll be annoying to work backwards when wanting more immediate errors, for lack of an is_err method on Query, but ok